jupyter-declarativewidgets-incorporation/jupyter-declarativewidgets-extension-incorporation.md

```diff
@@ -0,0 +1,181 @@
+* Reacting to changes in data on the Kernel, for example, new data arriving on a Stream.
+
+TODO: Need one or two animations of the markup and it's resulting UI
+
+### Current Features
+
+The following are the most relevant features:
+
+1. A set of declarative elements and corresponding kernel side code that enables connecting and interacting with functions, DataFrames and arbitrary data on the Kernel.
+2. A set of declarative elements for visualizing data in a variety of chart types.
+3. An extension to the `<template>` element that enables data bindings across multiple Notebook output cells.
+4. A notebook server extension to install and importing web-components at runtime.
+5. An implementation of all features for Python, R[[1]](#1), and Scala[[2]](#2) kernels.
+
+More details and information can be found in the [documentation](https://jupyter-incubator.github.io/declarativewidgets/docs.html). The project also contains many example and demo notebooks.
+
+### Design and Implementation
+
+The declarativewidgets extensions is a combination of:
+
+* A set of [web component](http://webcomponents.org/) elements and extensions built using [Polymer](https://www.polymer-project.org/1.0/)
+* A frontend extension that loads the elements and dependencies
+* A notebook server extension that enables installing web components using [Bower](https://bower.io/)
+* Kernel side implementations for Python, R[[1]](#1), and Scala[[2]](#2)
+
+#### Elements
+##### Core elements 
+
+* `<urth-core-function>` - Enable binding to functions defined in the kernel
+* `<urth-core-dataframe>` - Access to read and query DataFrames
+* `<urth-core-bind>` - Extension to the `template` tag to allow data binding across cell and with data on the kernel
+* `<urth-core-import>` - Extension to the `link` tag to allow installing and importing web components
+
+There are other `core` elements, but they play a supporting role.
+
+Some of the `core` elements are a combination of browser and kernel component. On the browser side, these elements are build on top of the `jupyter-js-widgets` foundation. They communicate with the kernel through the Comm channel[[3]](#3).
+
+##### Viz elements
+
+There is a collection of `<urth-viz-*` element that are pure client side and are use to display data. These elements include a table view and a variety of plots. 
+
+
+More details about the architecture can be found [here](https://github.com/jupyter-incubator/declarativewidgets/wiki). Additionally, see the [documentation](https://jupyter-incubator.github.io/declarativewidgets/docs.html).
+
+#### Notebook Server Extension
+
+The extension to the notebook server creates a new route for requesting the installation of new web component. The implementation uses Bower to download the packages for the web component and servers the content from a virtual path.
```

Comment:
Maybe a little blurb about "why bower?" -> "because that's what polymer ecosystem uses currently"